src/GenAPI/Microsoft.DotNet.GenAPI/INamedTypeSymbolExtension.cs

```diff
@@ -159,5 +160,56 @@ public static bool IsBoundGenericType(this INamedTypeSymbol namedType)
 
             return true;
         }
+
+        // Synthesize an internal default constructor for the type with implicit default constructor and the base type without.
+        public static IEnumerable<SyntaxNode> TryGetInternalDefaultConstructor(this INamedTypeSymbol namedType, ISymbolFilter symbolFilter)
+        {
+            if (!namedType.Constructors.Where(symbolFilter.Include).Any() && namedType.BaseType != null)
+            {
+                IEnumerable<IMethodSymbol> baseConstructors = namedType.BaseType.Constructors.Where(symbolFilter.Include);
+                if (baseConstructors.Any() && baseConstructors.All(c => !c.Parameters.IsEmpty))
+                {
+                    SyntaxKind visibility = SyntaxKind.InternalKeyword;
+
+                    Func<ISymbolFilter, bool> IncludeInternalSymbols = filter =>
+                        filter is AccessibilitySymbolFilter accessibilityFilter && accessibilityFilter.IncludeInternalSymbols;
+
+                    if (IncludeInternalSymbols(symbolFilter) ||
+                        (symbolFilter is CompositeSymbolFilter compositeSymbolFilter &&
+                            compositeSymbolFilter.Filters.Any(filter => IncludeInternalSymbols(filter))))
+                    {
+                        visibility = SyntaxKind.PrivateKeyword;
+                    }
+
+                    yield return SyntaxFactory.ConstructorDeclaration(
+                        new SyntaxList<AttributeListSyntax>(),
+                        SyntaxFactory.TokenList(new[] { SyntaxFactory.Token(visibility) }),
+                        SyntaxFactory.Identifier(namedType.ToDisplayString()),
+                        SyntaxFactory.ParameterList(),
+                        default!,
+                        default(BlockSyntax)!).WithInitializer(baseConstructors.First().GenerateBaseConstructorInitializer());
+                }
+            }
+        }
+
+        // Synthesize a base class initializer.
```

Comment:
As this API is publicly exposed, please use triple-slash documentation here: `///`

Reply:
Updated.
```
/// <summary>
/// Synthesize a base class initializer. 
/// </summary>
/// <param name="baseTypeConstructor">Represents a base class constructor <see cref="IMethodSymbol"/>.</param>
/// <returns></returns>
```